modules/setting/history/exposed_api.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package history
+
+// This file contains the methods that will be exposed to other packages
+import (
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting/base"
+)
+
+var removedSettings map[SettingsSource]map[string][]*historyEntry = make(map[SettingsSource]map[string][]*historyEntry) // ordered by old source and then by old section (for performance)
+
+func addHistoryEntry(entry *historyEntry) {
+	source := entry.oldValue.Source()
+	section := entry.oldValue.Section()
+	sections := removedSettings[source]
+
+	if sections == nil {
+		sections = make(map[string][]*historyEntry)
+		removedSettings[source] = sections
+	}
+
+	entriesInSection := sections[section]
+	entriesInSection = append(entriesInSection, entry)
+	sections[section] = entriesInSection
+}
+
+// MoveIniSetting adds a notice that the given setting under "[section].key" has been replaced by "[replacementSection].replacementKey" inside the ini configuration file
+// Everything should be exactly like it is in the configuration file
+func MoveIniSetting(version, section, key, replacementSection, replacementKey string) {
```

Comment:
Should all of these `MoveX` funcs be unexported? That would make it more clear that they are only to be used inside this package.

Reply:
These methods are explicitly built to be exposed (that's why this file is called `exposed_api`).
It is simply "architectural coincidence" that they are only used inside this package in the `Init`.
There is no reason at all to limit them to this package only, as their signature is not supposed to change anymore, and they don't rely on any package internal assumptions.
I built the API explicitly so that only the things that won't change are exported.

Reply:
Fair enough, I was just under the impression this PR was meaning to explicitly keep all changes in this list rather than scattered around like they were before.